tests/bin/notifications.test.ts

```diff
@@ -11,8 +11,18 @@ import { PolykeyAgent } from '@';
 import { makeVaultId } from '@/vaults/utils';
 import * as utils from './utils';
 import * as testUtils from './utils';
+import { pki } from "node-forge";
 // Import { makeVaultId } from "@/vaults/utils";
 
+// mocks.
+jest.mock('@/keys/utils', () => ({
```

Comment:
Here's an alternative way to do this:

Go to `tests/utils.ts`:

```ts
const keyPair = keysUtils.generateKeyPair(4096);

export { keyPair };
```

Now in each mock, can just do:

```ts
const generateDeterministicKeyPair = jest.fn().mockResolvedValue(testUtils.keyPair);
jest.mock('@/keys/utils', () => ({
  ...jest.requireActual('@/keys/utils'),
  generateDeterministicKeyPair
}));
```

Reply:
This ensures that the keypair is generated once for the entire set of tests.

Of course not all tests should have this mock applied.

Reply:
Since jest factories cannot refer to anything outside. Do the `const { keyPair } = require('../tests/utils');` inside the mock factory function.

Reply:
```ts
jest.mock('@/keys/utils', () => {
  const { keyPair } = require('../utils');
  return {
    ...jest.requireActual('@/keys/utils'),
    generateDeterministicKeyPair: jest.fn().mockResolvedValue(keyPair),
  };
});
```

Reply:
Jest allows one to do mocking globally too in `tests/setup.ts` or `tests/setupAfterEnv.ts` forgot the difference. However if you do this, this means you will never be testing the recovery code stuff. We at least do want to test that, so we only do mocking for specific modules.

Reply:
This has been changed, currently we're replacing `generateDeterministicKeyPair` with `generateKeyPair`